navierstokes.cpp

```diff
@@ -411,6 +435,33 @@ namespace Fluid
               }
           }
 
+        //Impose pressure boundary here, loop over faces on the cell
+        //and apply pressure boundary conditions
+        for (unsigned int face_n = 0;
+             face_n < GeometryInfo<dim>::faces_per_cell;
+             ++face_n)
+        {
+          if (cell->at_boundary(face_n) &&
+              parameters.fluid_neumann_bcs.find(cell->face(face_n)->boundary_id()) !=
+              parameters.fluid_neumann_bcs.end())
+          {
+            fe_face_values.reinit(cell,face_n);
+            unsigned int p_bc_id = cell->face(face_n)->boundary_id();
+            pressure_boundary_values[p_bc_id].value_list (fe_face_values.get_quadrature_points(),
```

Comment:
Since you already stored the `boundary id: pressure value` information as a map in `parameters.fluid_neumann_bcs`, and they are not likely to change at quadrature points, why not call `double p = parameters.fluid_neumann_bcs[p_bc_id]` instead of writing a `PressureBoundaryValues ` class?

Reply:
changed